Fix broken link to release notes in README (#15)
TonicAI/tonic_docker_compose: README.md

### Version

> - `VERSION_TAG`: "latest" or a specific version tag. Tonic's tag convention is just the release number, e.g. "123". Release notes are available at [docs.tonic.ai](https://www.tonic.ai/product-release-notes/structural).

Reviewer: ```suggestion
- `VERSION_TAG`: "latest" or a specific version tag. Tonic's tag convention is just the release number. For example, "123". Release notes are available on the [Tonic website](https://www.tonic.ai/product-release-notes/structural).
```
